File: typescript/user-interface/packages/ui-wasm/src/ts/gms-filters-provider/gms-filters/types/cascaded-filter-description.ts
import type { WasmCascadedFilterParameters } from './cascade-filter-parameters';
import type { WasmVectorFilterDescriptionWrapper } from './vector-filter-description-wrapper';

export interface WasmCascadedFilterDescriptionModule {
  new (
    causal: boolean,
    comments: string,
    filterDescriptions: WasmVectorFilterDescriptionWrapper,
    parameters: WasmCascadedFilterParameters
  ): WasmCascadedFilterDescription;
}

export interface WasmCascadedFilterDescription {
  $$: {
    ptr: number;
  };
  causal: boolean;
  comments: string;
  filterDescriptions: WasmVectorFilterDescriptionWrapper;
  parameters: WasmCascadedFilterParameters;
  delete: () => void;
}

File: cpp/gms/gms-filters-provider/src/descriptions/CascadedFilterDescription.cpp
#include "CascadedFilterDescription.hh"

CascadedFilterDescription::CascadedFilterDescription(bool causal,
                                                       std::string comments,
                                                       std::vector<FilterDescriptionWrapper> filterDescriptions,
                                                       CascadedFilterParameters parameters) : filterDescriptions(filterDescriptions),
                                                                                               parameters(parameters),
                                                                                               BaseFilterDescription(causal, comments){};